engine/pongscenes.cpp

```diff
@@ -0,0 +1,58 @@
+#include "pongscenes.h"
+
+#include "ball.h"
+#include "opponent.h"
+#include "player.h"
+#include "scorearea.h"
+#include "scorecontroller.h"
+#include "settingsscreencontroller.h"
+#include "titlescreencontroller.h"
+#include "wall.h"
+
+namespace pong
+{
+
+void TitleScene::BuildScene()
+{
+    GameObject* titleScreenController = NewObject();
```

Comment:
Does this allocate? looks like possibly a leak could be happening here

Reply:
Yeah it does allocate. The idea I had for these `BuildScene` functions would be that the scene would be created using helper functions to create objects and add them to the scene's internal object list. It would also return a pointer incase anything also needed to be done to the object, whether setting their position, name, or setting some Behavior member variable. Only after this Build function then the gameobject list will be moved and given to the main game loop to be properly used.

Reply:
I think this is good, fits in line with "raw pointers being non-owning.